core/domain/html_validation_service.py

```diff
@@ -705,3 +705,156 @@ def convert_svg_diagram_tags_to_image_tags(html_string):
+        (u'\xe2\u2026\u201d', u'\u2154'),
+        (u'\xe2\u2030\xa4', u'\u2264'),
+        (u'\xe2\u201a\xac', u'\u20ac'),
+        (u'\xe3\u201a\u201e', u'\u3084'),
+        (u'\xe3\u201a\u201c', u'\u3093'),
+        (u'\xe3\u201a\u201a', u'\u3082'),
+        (u'\xe3\u201a\u2019', u'\u3092'),
+        (u'\xe3\u201a\u0160', u'\u308a'),
+        (u'\xe5\u0152\u2014', u'\u5317'),
+        (u'\xe6\u0153\xa8', u'\u6728'),
+        (u'\xe6\u02c6\u2018', u'\u6211'),
+        (u'\xe6\u02dc\xaf', u'\u662f'),
+        (u'\xe8\xa5\xbf', u'\u897f'),
+        (u'\xe9\u201d\u2122', u'\u9519'),
+        (u'\u7aef', u'\xfc'),
+        (u'\u8d38', u'\xf3'),
+        (u'\ucc44', u'\xe4'),
+        (u'\uccb4', u'\xfc'),
+        (u'\u89ba', u'\u0131'),
+        (u'\uce74', u'\u012b'),
+        (u'\u0e23\u0e07', u'\xe7'),
+        (u'\u0e23\x97', u'\xd7'),
+        (u'\u0e23\u0e17', u'\xf7'),
+        (u'\u0e23\u0e16', u'\xf6'),
+        (u'\u0e23\u0e13', u'\xf3'),
+        (u'\u0e23\u0e1b', u'\xfb'),
+        (u'\xf0\u0178\u02dc\u2022', u'\U0001f615'),
+        (u'\xf0\u0178\u02dc\u0160', u'\U0001f60a'),
+        (u'\xf0\u0178\u02dc\u2030', u'\U0001f609'),
+        (u'\xf0\u0178\u2122\u201e', u'\U0001f644'),
+        (u'\xf0\u0178\u2122\u201a', u'\U0001f642'),
+        (u'\u011f\u0178\u02dc\u0160', u'\U0001f60a'),
+        (u'\u011f\u0178\u2019\xa1', u'\U0001f4a1'),
+        (u'\u011f\u0178\u02dc\u2018', u'\U0001f611'),
+        (u'\u011f\u0178\u02dc\u0160', u'\U0001f60a'),
+        (u'\xf0\u0178\u201d\u2013', u'\U0001f516'),
+        (u'\u011f\u0178\u02dc\u2030', u'\U0001f609'),
+        (u'\xf0\u0178\u02dc\u0192', u'\U0001f603'),
+        (u'\xf0\u0178\xa4\u2013', u'\U0001f916'),
+        (u'\xf0\u0178\u201c\xb7', u'\U0001f4f7'),
+        (u'\xf0\u0178\u02dc\u201a', u'\U0001f602'),
+        (u'\xf0\u0178\u201c\u20ac', u'\U0001f4c0'),
+        (u'\xf0\u0178\u2019\xbf', u'\U0001f4bf'),
+        (u'\xf0\u0178\u2019\xaf', u'\U0001f4af'),
+        (u'\xf0\u0178\u2019\xa1', u'\U0001f4a1'),
+        (u'\xf0\u0178\u2018\u2039', u'\U0001f44b'),
+        (u'\xf0\u0178\u02dc\xb1', u'\U0001f631'),
+        (u'\xf0\u0178\u02dc\u2018', u'\U0001f611'),
+        (u'\xf0\u0178\u02dc\u0160', u'\U0001f60a'),
+        (u'\xf0\u0178\u017d\xa7', u'\U0001f3a7'),
+        (u'\xf0\u0178\u017d\u2122', u'\U0001f399'),
+        (u'\xf0\u0178\u017d\xbc', u'\U0001f3bc'),
+        (u'\xf0\u0178\u201c\xbb', u'\U0001f4fb'),
+        (u'\xf0\u0178\xa4\xb3', u'\U0001f933'),
+        (u'\xf0\u0178\u2018\u0152', u'\U0001f44c'),
+        (u'\xf0\u0178\u0161\xa6', u'\U0001f6a6'),
+        (u'\xf0\u0178\xa4\u2014', u'\U0001f917'),
+        (u'\xf0\u0178\u02dc\u201e', u'\U0001f604'),
+        (u'\xf0\u0178\u2018\u2030', u'\U0001f449'),
+        (u'\xf0\u0178\u201c\xa1', u'\U0001f4e1'),
+        (u'\xf0\u0178\u201c\xa3', u'\U0001f4e3'),
+        (u'\xf0\u0178\u201c\xa2', u'\U0001f4e2'),
+        (u'\xf0\u0178\u201d\u0160', u'\U0001f50a'),
+        (u'\xc2\xb2', u'\xb2'),
```

Comment:
Please replace everything including this line and below with:

```
        (u'\xc2', ''),
        (u'\t', u''),
        (u'\n', u''),
        (u'\xa0', u' '),
        (u'&nbsp;', u' '),
```

Here are the rationales:
- Some old strings contain \xc2 which is being dropped from the new.
- Some old strings contain \t and \n, this should be removed. They have been replaced by an empty string already and there's no way to recover the old characters anyway.
- Some old strings contain \xa0 which has been replaced inconsistently by either &nbsp; or a space in the new strings. It is not possible to recover these, so we drop all '\xa0's and change them to spaces.

Reply:
I've updated based on the latest list!